tests/FilterTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Tests;
+
+use Illuminate\Http\Request;
+use Mockery;
+use OveD\Apm\Filters\NoOptionRequests;
+use OveD\Apm\Filters\Sampling;
+
+class FilterTest extends ApmTestCase
+{
+
+    /** @test */
+    public function can_filter_options_requests()
+    {
+        // Given
+        $filter = new NoOptionRequests();
+
+        // When
+        $mock = \Mockery::mock(Request::class);
+        $mock->shouldReceive('method')->andReturn('OPTIONS');
+        $reject = $filter->shouldReject($mock);
+
+        // Then
+        $this->assertEquals(true, $reject);
+    }
+
+    /** @test */
+    public function can_turn_off_apm()
+    {
+        // Given
+        $sampling = new Sampling(0);
+
+        // When
+        $mock = Mockery::mock(Request::class);
+        for ($i =0; $i < 10000; $i++){
+            $this->assertFalse($sampling->shouldReject($mock));
+        }
+        // Then
+    }
+
+    /** @test */
+    public function can_turn_on_apm()
+    {
+        // Given
+        $sampling = new Sampling(100);
+
+        // When
+        $mock = Mockery::mock(Request::class);
+        for ($i =0; $i < 10000; $i++){
+            $this->assertTrue($sampling->shouldReject($mock));
+        }
+        // Then
```

Comment:
What is this testing exactly?

Reply:
That if you take 10000 samples at 100% all of them will return true. I had a bug at a point where it would miss a few even with 100%

Reply:
The name of the test is `can_turn_on_apm`... That's what's confusing me :)

Reply:
Ah you turn it on/off with sampling... That was a bit unclear